src/System.Linq/tests/ConcatTests.cs

```diff
@@ -251,20 +251,34 @@ public void CountOfConcatIteratorShouldThrowExceptionOnIntegerOverflow()
             var supposedlyLargeCollection = new DelegateBasedCollection<int> { CountWorker = () => int.MaxValue };
             var tinyCollection = new DelegateBasedCollection<int> { CountWorker = () => 1 };
 
+            Action<Action> assertThrows = (testCode) =>
+            {
+                // The full .NET framework uses unsigned arithmetic summing up collection counts.
+                // See https://github.com/dotnet/corefx/pull/6131.
+                if (PlatformDetection.IsFullFramework)
+                {
+                    testCode();
+                }
+                else
+                {
+                    Assert.Throws<OverflowException>(testCode);
```

Comment:
I think this change in behavior is ok. 
Even if Count does not throw on desktop, it must be reporting wrong value, when result does not fit in int32.
The part where ToArray() may work in these case is only because the collections report invalid Count. If they really had that many items, ToArray would fail.

I think none of the cases can be considered as even remotely "normal", so this does not look like breaking.